src/clarity_vm/database/mod.rs

```diff
@@ -224,6 +227,72 @@ fn get_matured_reward(conn: &DBConn, child_id_bhh: &StacksBlockId) -> Option<Min
     }
 }
 
+/// This trait describes SortitionDB connections. This is used
+/// for methods that the chainstate needs to be in common between
+/// different sortition db connections or handles, but that aren't
+/// used by the `clarity_db` (and therefore shouldn't appear in BurnStateDB)
+pub trait SortitionDBRef: BurnStateDB {
+    fn get_pox_start_cycle_info(
+        &self,
+        sortition_id: &SortitionId,
+        parent_stacks_block_burn_ht: u64,
+        cycle_index: u64,
+    ) -> Result<Option<PoxStartCycleInfo>, ChainstateError>;
+}
+
+fn get_pox_start_cycle_info(
+    handle: &mut SortitionHandleConn,
+    parent_stacks_block_burn_ht: u64,
+    cycle_index: u64,
+) -> Result<Option<PoxStartCycleInfo>, ChainstateError> {
+    let descended_from_last_pox_anchor = match handle.get_last_anchor_block_hash()? {
+        Some(pox_anchor) => handle.descended_from(parent_stacks_block_burn_ht, &pox_anchor)?,
+        None => return Ok(None),
+    };
+
+    if !descended_from_last_pox_anchor {
```

Comment:
Wait, there's no reward addresses for PoX if miners choose an anchor block that's the nth ancestor of the last-chosen anchor block?  Is this part of SIP-007?

Reply:
I'm not sure I understand this question. This method is attempting to fetch the information stored at the start of the current PoX reward cycle if and only if the current block descends from the anchor block. So, if a PoX anchor block is chosen, if this Stacks block is in a fork that doesn't build off of that anchor block, then it cannot process information for that anchor block.

Reply:
Gotcha. I was thrown off by the variable name `descended_from_last_pox_anchor`.  I thought this meant "last PoX anchor that was selected anywhere" (which is wrong) vs. "last PoX anchor selected in this sortition history".